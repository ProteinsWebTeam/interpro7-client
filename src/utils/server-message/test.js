// @flow
import ShallowRenderer from 'react-test-renderer/shallow';

import getServerMessage from '.';

describe('server-message', () => {
  test('should render', () => {
    // eslint-disable-next-line no-magic-numbers
    for (const status of [204, 400, 404, 408, 500]) {
      expect(getServerMessage(status)).toMatchSnapshot();
    }
  });

  // `undefined` if OK, or not a valid status
  test("shouldn't render", () => {
    // eslint-disable-next-line no-magic-numbers
    for (const status of [200, 999]) {
      expect(getServerMessage(status)).toBeUndefined();
    }
  });
});
