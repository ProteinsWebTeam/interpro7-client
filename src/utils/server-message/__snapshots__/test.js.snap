// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server-message should render 1`] = `
<div
  className="callout withicon info"
>
  Valid query, but unfortunately no data was available on the server
</div>
`;

exports[`server-message should render 2`] = `
<div
  className="callout withicon alert"
>
  There was an error, this is not a valid request to the server
</div>
`;

exports[`server-message should render 3`] = `
<div
  className="callout withicon alert"
>
  There was an error, this is not a known resource on the server
</div>
`;

exports[`server-message should render 4`] = `
<div
  className="callout withicon alert"
>
  The query is still running in the background. We will update the page once the data is ready
</div>
`;

exports[`server-message should render 5`] = `
<div
  className="callout withicon alert"
>
  There was an error while getting data from the server
</div>
`;
